lib/compiler/src/engine/artifact.rs

```diff
@@ -573,6 +593,16 @@ impl Artifact {
         ))
     }
 
+    fn get_byte_slice(input: &[u8], start: usize, end: usize) -> Option<&[u8]> {
```

Comment:
This can just return a Result and then the code will be greatly simplified:

```
.ok_or(DeserializeError::InvalidByteLength {
                    expected: cur_offset + WORD_SIZE,
                    got: bytes.len(),
                })
```